python/lsst/qa/explorer/classifyObjects.py

```diff
@@ -0,0 +1,371 @@
+# This file is part of qa explorer
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Applys star galaxy classifier
+"""
+
+import lsst.pex.config as pexConfig
+from lsst.pipe.base import Task
+from .parquetTable import ParquetTable
+
+import numpy as np
+import pickle
+
+
+class StarGalaxyClassifierConfig(pexConfig.Config):
+    """Config for star galaxy classifier
+    """
+    version = pexConfig.Field(
+        doc="Version information",
+        dtype=str,
+        default="HSC-COSMOS-20180528",
+    )
+
+    pass
+
+
+class StarGalaxyClassifierTask(Task):
+    """Add classification (star or galaxy) and associated flag columns to a parquet table.
+    """
+
+    _DefaultName = "starGalaxyClassifier"
+    ConfigClass = StarGalaxyClassifierConfig
+
+    def __init__(self):
+        Task.__init__(self)
+
+    def runDataRef(self, dataRef):
+        """Add required magnitudes and run the classifier
+
+        checks that all the required filters are in the file, then adds magntiudes and runs the classifier
+        for the given patch and tract.
+        """
+        cat = dataRef.get("deepCoadd_obj").toDataFrame({"dataset": "meas"})
+
+        # For now raise a not implemented error for other cameras, eventually needs other trained classifiers
+        camera_name = dataRef.getButler()._getDefaultMapper().getCameraName()
```

Comment:
I suggest `dataRef.ge("camera").getName()` because it uses public methods and is closer to something that will work with the gen3 butler.